HeterogeneousCore/SonicCore/src/SonicClientBase.cc

```diff
@@ -40,24 +55,30 @@ void SonicClientBase::start(edm::WaitingTaskWithArenaHolder holder) {
   holder_ = std::move(holder);
 }
 
-void SonicClientBase::start() { tries_ = 0; }
+void SonicClientBase::start() {
+  totalTries_ = 0;
+  // initialize all actions
+  for (const auto& action : retryActions_) {
```

Comment:
this should probably actually be just `auto&` with no `const`. The underlying action has non-const methods that can be called during this process. `const auto& action` is really just the `unique_ptr`, and this may be confusing for understanding the `const`-ness of the overall loop. (the same should be applied to other loops over `retryActions_`)

Reply:
Done.